components/automate-cds/server/root.go

```diff
@@ -46,63 +59,87 @@ func (s *Server) GetVersion(
+		}
+	}
 
-	// Canned data
 	return &response.ContentItems{
-		Items: []*response.ContentItem{
-			{
-				Id:             "f354577e-2ea5-4c96-86ed-78a8ba68e0f3",
-				Name:           "DevSec Linux Security Baseline Benchmark Level 1",
-				Description:    "Test suite for best practice Linux OS hardening",
-				Type:           "profile",
-				Version:        "1.9.0",
-				Platforms:      []string{"Ubuntu", "Debian"},
-				CanBeInstalled: true,
-				Filename:       "linux-baseline-2.2.2.tar.gz",
-			},
-			{
-				Id:             "1f9e6a1e-3381-487f-98ef-9966c65ff92a",
-				Name:           "Audit Benchmark level 1",
-				Description:    "Test suite for best practice Linux OS hardening",
-				Type:           "cookbook",
-				Version:        "9.2.1",
-				Platforms:      []string{"Redhat", "Centos"},
-				CanBeInstalled: false,
-				Filename:       "audit-benchmark-level-1.tar.gz",
-			},
-			{
-				Id:             "736f5724-b384-4d74-b8cb-dedd9f1ecd54",
-				Name:           "Compliance Effortless Package",
-				Description:    "This is a hart file that enables effortless compliance on your fleet.",
-				Type:           "package",
-				Version:        "2.3.2",
-				Platforms:      []string{"Solaris"},
-				CanBeInstalled: true,
-				Filename:       "chef-compliance-effortless-1.1.1-20200626161151-x86_64-linux.hart",
-			},
-		},
+		Items: responseItems,
 	}, nil
 }
 
 // InstallContentItem - installing a content item
-func (s *Server) InstallContentItem(ctx context.Context, request *request.InstallContentItem) (*response.InstallContentItem, error) {
+func (s *Server) InstallContentItem(ctx context.Context,
+	request *request.InstallContentItem) (*response.InstallContentItem, error) {
+	contentItem, found, err := s.client.GetContentItem(request.Id)
+	if err != nil {
+		return &response.InstallContentItem{}, status.Errorf(codes.Internal,
+			"Could not connect to Content Delivery Service error: %s", err.Error)
+	}
+	if !found {
+		return &response.InstallContentItem{}, status.Errorf(codes.InvalidArgument,
+			"Content item not found with ID %q", request.Id)
+	}
 
-	log.Infof("Downloading content item with ID %s ...", request.Id)
+	if !contentItem.CanBeInstalled {
+		return &response.InstallContentItem{}, status.Errorf(codes.InvalidArgument,
+			"Content item with ID %q Can not be installed", request.Id)
+	}
 
-	log.Infof("Installing content item with ID %s ...", request.Id)
+	if contentItem.Type == "profile" {
+		if request.RequestUser == "" {
+			return &response.InstallContentItem{}, status.Errorf(codes.InvalidArgument,
+				"A request_user must be non empty")
+		}
+
+		err = s.installProfile(ctx, contentItem, request.RequestUser)
+		if err != nil {
+			return &response.InstallContentItem{}, status.Errorf(codes.Internal,
+				"Error Installing Profile: %s", err.Error)
+		}
+	} else {
+		return &response.InstallContentItem{}, status.Errorf(codes.InvalidArgument,
+			"Can install content item type %q", contentItem.Type)
+	}
```

Comment:
Filled out the install function more to provide errors for content items that can not be installed.